packages/cli/src/config/index.ts

```diff
@@ -1,28 +1,55 @@
+/* eslint-disable @typescript-eslint/naming-convention */
 /* eslint-disable @typescript-eslint/unbound-method */
 /* eslint-disable no-console */
 import convict from 'convict';
 import dotenv from 'dotenv';
+import { tmpdir } from 'os';
+import { mkdtempSync } from 'fs';
+import { join } from 'path';
 import { schema } from './schema';
 
-dotenv.config();
+const inE2ETests = process.env.E2E_TESTS === 'true';
+
+if (inE2ETests) {
+	// Skip loading config from env variables in end-to-end tests
+	process.env = {
+		N8N_USER_FOLDER: mkdtempSync(join(tmpdir(), 'n8n-e2e-')),
+		N8N_DIAGNOSTICS_ENABLED: 'false',
+		N8N_PUBLIC_API_DISABLED: 'true',
+		EXTERNAL_FRONTEND_HOOKS_URLS: '',
+		N8N_PERSONALIZATION_ENABLED: 'false',
+	};
+} else {
+	dotenv.config();
+}
 
 const config = convict(schema);
 
+if (inE2ETests) {
+	config.set('enterprise.features.sharing', true);
+	config.set('enterprise.workflowSharingEnabled', true);
```

Comment:
Would be nice to be able to configure this using a cypress function in the future.